percy/providers/app_automate.py

```diff
@@ -15,9 +15,11 @@ def supports(remote_url) -> bool:
     def screenshot(self, name: str, **kwargs):
         session_details = self.execute_percy_screenshot_begin(name)
         # Device name and OS version retrieval is custom for App Automate users
-        self.metadata._device_name = kwargs.get('device_name') or session_details.get("deviceName")
-        self.metadata._os_version = session_details.get("osVersion")
-        self.set_debug_url(session_details)
+        if session_details is not None:
+            self.metadata._device_name = kwargs.get('device_name') or session_details.get("deviceName")
```

Comment:
if user has provided device_name in kwargs and if we get session_details none, then this will not set because of this check

Reply:
Its handled in get tag method. This SDK is slightly out of sync with other SDKs